### DISCUSSION TITLE
craft setup/keys --force (Add "force" flag)

### DISCUSSION THREAD

**Discussion by lindseydiloreto** [craftcms/cms#13165] [Ideas] [answered]
**TL;DR** - It would be great if `craft setup/keys` supported a `--force` flag, to ignore and overwrite any pre-existing values.

---

Within the `craft` CLI, the `setup/keys` command is great! It generates both an application ID and security key _(assuming they don't already exist)_.

However, those values are **not empty** in our starter project. It contains default values for both...

```env
# The application ID used to to uniquely store session and cache data, mutex locks, and more
CRAFT_APP_ID="./craft setup/app-id"

# The secure key Craft will use for hashing and encrypting data
CRAFT_SECURITY_KEY="./craft setup/security-key"
```

^ These values give us an easy detour, if for some reason the project is configured outside of our normal routine. In a pinch, we can just copy & paste those commands to generate the _real_ info.

Here's the problem... When we spin up a new project using `composer create-project`, the `post-create-project-cmd` command runs a script that (ideally) includes `setup/keys`. This script has been working great using `setup/app-id` and `setup/security-key` separately, I'd love to combine them if possible.

**Would it be possible to add a `--force` flag to the `setup/keys` command?**

**Comment by brandonkelly** [comment, marked as answer]
You can run `setup/app-id` and `setup/security-key` individually, and new keys will be generated regardless of whether there’s existing values. Seems like adding the second call is basically about the same amount of work as adding `--force`, so not sure the additional arg is worth it.

**Comment by lindseydiloreto** [comment]
Alright, figured I'd ask. I'll just stick with the original two separate commands (which have worked great so far).
